plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergStatistics.java

```diff
@@ -0,0 +1,328 @@
+    private final Map<Integer, Long> columnSizes;
+
+    private IcebergStatistics(
+            long recordCount,
+            long fileCount,
+            long size,
+            Map<Integer, Object> minValues,
+            Map<Integer, Object> maxValues,
+            Map<Integer, Long> nullCounts,
+            Map<Integer, Long> columnSizes)
+    {
+        this.recordCount = recordCount;
+        this.fileCount = fileCount;
+        this.size = size;
+        this.minValues = requireNonNull(minValues, "minValues is null");
+        this.maxValues = requireNonNull(maxValues, "maxValues is null");
+        this.nullCounts = requireNonNull(nullCounts, "nullCounts is null");
+        this.columnSizes = requireNonNull(columnSizes, "columnSizes is null");
+    }
+
+    public long getRecordCount()
+    {
+        return recordCount;
+    }
+
+    public long getFileCount()
+    {
+        return fileCount;
+    }
+
+    public long getSize()
+    {
+        return size;
+    }
+
+    public Map<Integer, Object> getMinValues()
+    {
+        return minValues;
+    }
+
+    public Map<Integer, Object> getMaxValues()
+    {
+        return maxValues;
+    }
+
+    public Map<Integer, Long> getNullCounts()
+    {
+        return nullCounts;
+    }
+
+    public Map<Integer, Long> getColumnSizes()
+    {
+        return columnSizes;
+    }
+
+    public static class IcebergStatisticsBuilder
+    {
+        private final Map<Integer, Type.PrimitiveType> idToTypeMapping;
+        private final List<Types.NestedField> columns;
+        private final TypeManager typeManager;
+        private final Map<Integer, Optional<Long>> nullCounts = new HashMap<>();
+        private final Map<Integer, ColumnStatistics> columnStatistics = new HashMap<>();
+        private final Map<Integer, Long> columnSizes = new HashMap<>();
+        private long recordCount;
+        private long fileCount;
+        private long size;
+
+        public IcebergStatisticsBuilder(
+                Map<Integer, Type.PrimitiveType> idToTypeMapping,
+                List<Types.NestedField> columns,
+                TypeManager typeManager)
+        {
+            this.idToTypeMapping = requireNonNull(idToTypeMapping, "idToTypeMapping is null");
+            this.columns = requireNonNull(columns, "columns is null");
+            this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        }
+
+        public void registerDataFile(DataFile dataFile, PartitionSpec partitionSpec)
+        {
+            fileCount++;
+            recordCount += dataFile.recordCount();
+            size += dataFile.fileSizeInBytes();
+
+            Map<Integer, Long> newColumnSizes = dataFile.columnSizes();
+            if (newColumnSizes != null) {
+                for (Types.NestedField column : columns) {
+                    int id = column.fieldId();
+                    Long addedSize = newColumnSizes.get(id);
+                    if (addedSize != null) {
```

Comment:
you can `getOrDefault` to avoid null check.

Reply:
Do you know if a `null` column size is treated differently from a `0L` data size? I'm just thinking that default might have an impact on CBO

Reply:
good point, it’s different (no info vs zero-length)

Reply:
but, if you don't have `addedSize` for some files, and you do for some files, you produces too small a value now.
it could be reasonable to average size per row for files where this info is available, and hen multiply by number of rows

Reply:
Note to self, didn't do this yet.

Reply:
Through, for what it's worth. That's how it worked before: https://github.com/trinodb/trino/pull/9906/files#diff-a07b03293b2a177bcd81417b6e96930d2cfd28c153c5be6857aa3d997ada1f82L291